### PR TITLE
Image service upgrade && max registrations in event

### PR DESCRIPTION
**Pull request by ritiksr25**
Image Service:
- Added filefilters for image
- using aws-sdk for upload instead of multer-s3

Max Registrations:
- event has limited seats for participants
- can be updated by lead
- relevant validations/ updates added

### CODE REVIEW

File: config/imageService.js [hunk 1]
```diff
@@ -54,6 +46,6 @@ module.exports.deleteImage = async key => {
 	} catch (err) {
 		console.log(err);
 		logger("error", "imageService", err);
-		throw err;
```
Comment (rhnmht30) [inline config/imageService.js]: Why is this remove? If we don't throw the error here then the try-catch surrounding deleteImg will not catch the error

File: controllers/events_controller.js [hunk 2]
```diff
@@ -399,11 +419,17 @@ module.exports.addEvent = async (req, res) => {
...
+		maxRegister
 	});
 
-	if (req.files && req.files.length !== 0 ) {
-		event.image = req.files[0].location;
+	if (req.files && req.files.length !== 0) {
+		let file = req.files[0];
+		let key = getImageKey(req.originalUrl);
+		let uploaded = await uploadImage(file, key);
+		if (uploaded) {
```
Comment (rhnmht30) [inline controllers/events_controller.js]: Will the variable needs to be checked only for true/false values or should we also check some other parameters? What is it returning specifically?

File: config/imageService.js [hunk 3]
```diff
@@ -16,31 +14,25 @@ aws.config.region = AWS_REGION;
...
-});
+		ServerSideEncryption: "AES256"
+	};
+	try {
+		let resp = await uploadObjectAsync(params);
+		return resp.Location;
+	} catch (err) {
+		console.log(err);
+		logger("error", "imageService", err);
+		return;
```
Comment (rhnmht30) [inline config/imageService.js]: I think here also we need to throw the error

File: config/imageService.js (hunk 1 above)
Comment (ritiksr25) [reply config/imageService.js]: its not required as it should not block the update as it's server issue, not at client side, so if by chance, if an image is not deleted, we are not bothered, correct me if something wrong!

File: controllers/events_controller.js (hunk 2 above)
Comment (ritiksr25) [reply controllers/events_controller.js]: if image is uploaded, it gives the URL, else undefined

File: config/imageService.js (hunk 1 above)
Comment (rhnmht30) [reply config/imageService.js]: Cul

**Pull request merged**